crates/sui-rest-api/src/transactions/resolve/mod.rs

```diff
@@ -388,130 +418,337 @@ fn resolve_arg(
     arg: UnresolvedInputArgument,
     arg_idx: usize,
 ) -> Result<CallArg> {
-    match arg {
-        UnresolvedInputArgument::Pure { value } => CallArg::Pure(value),
-        UnresolvedInputArgument::ImmutableOrOwned(obj_ref) => CallArg::Object(
-            ObjectArg::ImmOrOwnedObject(resolve_object_reference(reader, obj_ref)?),
+    use fastcrypto::encoding::Base64;
+    use fastcrypto::encoding::Encoding;
+    use sui_sdk_types::types::UnresolvedInputArgumentKind::*;
+
+    let UnresolvedInputArgument {
+        kind,
+        value,
+        object_id,
+        version,
+        digest,
+        mutable,
+    } = arg;
+
+    match (kind, value, object_id, version, digest, mutable) {
+        // pre serialized BCS input encoded as a base64 string
+        (Some(Pure), Some(UnresolvedValue::String(v)), None, None, None, None) => {
+            let value = Base64::decode(&v).map_err(|e| {
+                RestError::new(
+                    axum::http::StatusCode::BAD_REQUEST,
+                    format!("argument is an invalid pure arguement: {e}"),
+                )
+            })?;
+            CallArg::Pure(value)
+        }
+        // pre serialized BCS input encoded as a a JSON array of u8s
+        (Some(Pure), Some(array @ UnresolvedValue::Array(_)), None, None, None, None) => {
+            let value = serde_json::from_value(serde_json::Value::from(array)).map_err(|e| {
+                RestError::new(
+                    axum::http::StatusCode::BAD_REQUEST,
+                    format!("argument is an invalid pure arguement: {e}"),
+                )
+            })?;
+            CallArg::Pure(value)
+        }
+
+        // Literal, unresolved pure argument
+        (Some(Literal), Some(value), None, None, None, None)
+        | (None, Some(value), None, None, None, None) => CallArg::Pure(literal::resolve_literal(
+            called_packages,
+            commands,
+            arg_idx,
+            value,
+        )?),
```

Comment:
Note that in the design of the Rust SDK, we also wanted to support passing an ObjectID as a literal (string containing an address), and inferring based on usage that it is in fact an Object argument.